Sources/Swift/Integrations/SessionReplay/SentryOnDemandReplay.swift

```diff
@@ -158,13 +165,14 @@ class SentryOnDemandReplay: NSObject {
         
         var start = dateProvider.date()
         var actualEnd = start
-        
-        for frame in frames {
-            if frame.time < beginning { continue } else if frame.time > end { break }
-            if frame.time < start { start = frame.time }
-            
-            actualEnd = frame.time
-            framesPath.append(frame.imagePath)
+        workingQueue.dispatchSync {
```

Comment:
`m`: When this is called from the main thread, this won't work, if I'm not mistaken, because we access the frames on the main thread, but they could be modified on a background thread in the meantime.

Reply:
So, yes and no.

If we call this from the main thread, it will be executed in the main thread but only after all other dispatch code finishes execution. Moreover, if we call any asynchronous code while this one is running, the new one will have to wait.

Once again, the dispatch queue only executes code serially

Reply:
Oh, SentryDispatchSync is not the same as DispatchQueueSync 😒

I will add a true sync function

Reply:
> Oh, SentryDispatchWrapper.dispatchSync is not the same as DispatchQueue.sync 😒

I think we must rename `SentryDispatchWrapper.dispatchSync` to avoid confusion.

Reply:
Ah you added a new true sync function. You must not rename this one. Sorry didn't see it.